### PR TITLE
fix process pending consolidations

### PR DESCRIPTION
**Pull request by g11tech**
fix the consolidation bug in devnet 2

tl-dr of the incident:

a consolidation was initiated for a validator 106 into 107 and got accepted with withdrawal epoch (consolidation epoch 1386) 
image

so it was to be consolidated on epoch transition before slot 1386*32= 44352

but it was not processed untill the next epoch transition before the slot 1387*32 = 44384

in the mean time before 1386 and 1387 epoch, it was fully withdrawn in slot: `44359` because of process withdrawals
image

so ideally it should have been processed on 1386 epoch transition but was not because of this condition:
```python
 if source_validator.withdrawable_epoch > get_current_epoch(state):
            break
```
where epoch of state still comes out to be 1385 because state's slot has not yet ticked to `44352` which happens in processSlot which follows process epoch

full context: https://discord.com/channels/595666850260713488/1268911253913337897/1270009116571472065

this was the second consolidation which exhibited the same issue, previous one was triggered to consolidate 105 into 106 which again was not processed at its `"withdrawable_epoch": "697"` but 1 epoch later

### CODE REVIEW

**Comment by dapplion** [comment]
Can you summarize the context beside the link? 🙏

**Comment by g11tech** [comment]
> Can you summarize the context beside the link? 🙏

done

**Comment by g11tech** [comment]
closing in favor of https://github.com/ethereum/consensus-specs/pull/3868
